geode-junit/src/main/java/org/apache/geode/test/junit/rules/gfsh/GfshRule.java

```diff
@@ -145,9 +145,7 @@ public GfshExecution execute(GfshScript gfshScript, File workingDir) {
 
   public GfshExecution execute(GfshScript gfshScript) {
     try {
-      File workingDir = new File(temporaryFolder.getRoot(), gfshScript.getName());
-      workingDir.mkdirs();
-      return execute(gfshScript, workingDir);
+      return execute(gfshScript, temporaryFolder.getRoot());
```

Comment:
Note: The workingDir for gfshScript is the workingDir for the GFSH process spawned by the JUnit test. GFSH then spawns the Locator and Server which is running in a subfolder. This change makes it easy for GfshRule to stop ALL Locators and Servers anywhere in the TemporaryFolder. Since only one gfshScript is executed at a time, the GFSH workingDir should be ok as temporaryFolder.getRoot(). 

On the off-chance that this does cause a problem, it would have to be for other GFSH commands that write something to the current workingDir of GFSH. I'm fairly confident we don't have tests that will have disk artifacts colliding just because of this little change.

Reply:
so basically all scripts executed under the same gfshRule would use the same gfsh working dir instead of each execution has its own. I think this is a good change.

Reply:
Looks like your change made some other tests that uses gfshRule fail. Can I offer a different approach to fix your test only? This way you don't need to touch other tests and the gfshRule. I had one of your tests modified and put it on my repo so that it can compare with yours:

https://github.com/kirklund/geode/commit/e88006e8f1c9f5924f7daacbc1a7ef5668d72212